src/app/api/v1/admin/feature-flags/[flagId]/route.ts

```diff
@@ -28,3 +35,41 @@ export async function GET(
     return NextResponse.redirect(appConstants.SERVER_URL, 301);
   }
 }
+
+export async function POST(req: NextRequest) {
+  const session = await getServerSession(authOptions);
+  if (isAdmin(session?.user?.email || "")) {
+    // Signed in
+    try {
+      const flagName = req.nextUrl.pathname.split("/").at(-1);
+      if (!flagName) {
+        throw new Error("No flag name provided");
+      }
+      const result = await req.json();
+
+      if (result.id === "isEnabled") {
+        await enableFeatureFlagByName(flagName, result.isEnabled);
+      } else if (result.id === "dependencies") {
+        const dependencies = result.value.split(",");
+        await updateDependencies(flagName, dependencies);
+      } else if (result.id === "allowList") {
+        const allowList = result.value.split(",");
+        await updateAllowList(flagName, allowList);
+      } else if (result.id === "waitList") {
+        const waitList = result.value.split(",");
+        await updateWaitList(flagName, waitList);
+      } else if (result.id === "owner") {
+        const owner = result.value;
+        await updateOwner(flagName, owner);
+      }
+
+      return NextResponse.json({ success: true }, { status: 200 });
+    } catch (e) {
+      console.error(e);
+      return NextResponse.json({ success: false }, { status: 500 });
+    }
+  } else {
+    // Not Signed in, redirect to home
+    return NextResponse.redirect(appConstants.SERVER_URL, 301);
+  }
```

Comment:
tiniest of nits, and not sure if we have a style guide, but would it be easier to invert the `if..else` and just abort early if the user is not auth'ed and reduce a level of nesting?

```js
  const session = await getServerSession(authOptions);
  if (!isAdmin(session?.user?.email || "")) {
    // Not Signed in, redirect to home
    return NextResponse.redirect(appConstants.SERVER_URL, 301);
  }

  // Signed in
  try {
    const flagName = req.nextUrl.pathname.split("/").at(-1);
    ...
```

Reply:
I think "whatever `prettier` does, unless we override it" is our de factor style guide :P But yes I think you're right that earlier return is a better pattern, thanks.

Reply:
Only because I mentioned Next.js middleware and started googling: https://nextjs.org/docs/app/building-your-application/routing/middleware

Not sure if that's something we'd want to poke at (although not in this PR, just... in general). Basically create a middleware that looks for the `/(authenticated)/` route group [if that's possible` and auto-redirect to the homepage if they're not logged in so we don't need to do that in _n_ places.

Reply:
Yes I had the same question... I'd like to be able to be able to just trust the authenticated route to do the right thing and not have redundant checks everywhere.

Reply:
One problem is that we keep the APIs in `./src/app/api` and each endpoint implementation does it's own checking, I wonder if we could simplify there as well.

Reply:
Oh, and another thing - we need to check not only that the user is authenticated but *also* that they are using an admin account, I wonder if this is something we could do using middleware as well (and how we could test to make sure things aren't misconfigured and leaving the route open to normal users etc)

Reply:
Yeah, if we can do middleware for everything in `(authenticated)` maybe we create another route group for `(admin)` which adds bonus middleware for being both logged in AND part of some blessed admin group.

I believe FxA recently added a bunch of extended role based permissions for admin actions last year: https://mozilla.github.io/ecosystem-platform/reference/admin-panel#features-guarded-by-permissions 

I also think their https://github.com/mozilla/fxa/blame/main/packages/fxa-admin-panel/src/components/Guard/index.tsx system is protected by being both (a) behind VPN, and (b) accessible to only certain LDAP groups. But maybe that's right overkill for experiments if we only want to use some binary "isAdmin" flag in the database and not more granular roles like "can create feature flags" and "can delete feature flags" roles.

Reply:
I looked at a couple of tutorials for auth middleware, couldn't get any of them to work in our context. I agree it's definitely something we should look into, we definitely shouldn't have these if/else everywhere for session user checking